js/src/hooks/useNavigateAwayPromptEffect.js

```diff
@@ -18,27 +18,45 @@ export default function useNavigateAwayPromptEffect(
 	blockedLocation = () => true
 ) {
 	useEffect( () => {
-		// Bind beforeunload event for non `woocommerce/navigation` links and reloads.
-		// history#v5 does bind `beforeunload` automatically, with v4 we need to do it ourselves.
 		const eventListener = ( e ) => {
 			// If you prevent default behavior in Mozilla Firefox prompt will always be shown.
 			e.preventDefault();
 			// Chrome requires returnValue to be set.
 			e.returnValue = message;
 		};
 
-		if ( shouldBlock ) {
-			window.addEventListener( 'beforeunload', eventListener );
-		}
-		// Block woocommerce/navigation.
-		const unblock = getHistory().block( ( location ) => {
-			// TODO: Consider being futureproof for history.v5 where
-			// getHistory().block( ( { location, retry } ) => {
+		// Block woocommerce/navigation in order to show a confirmation prompt in case shouldBlock is true
+		const unblock = getHistory().block( ( transition ) => {
+			let shouldUnblock = true;
 
-			// Return the message / truthy value to show a prompt.
+			if ( shouldBlock ) {
+				window.addEventListener( 'beforeunload', eventListener );
+			}
```

Comment:
By moving `addEventListener` inside you lost the initial case

### Steps to reproduce
(exactly as in https://github.com/woocommerce/google-listings-and-ads/pull/1580#issuecomment-1176453957)
1. Go to edit free listings page `/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard&subpath=%2Ffree-listings%2Fedit`
2. Do a change
3. Click in browser's chrome to close the tab or reload the page
4. :red_circle: You should get the confirmation prompt


## What works
1. Go to edit free listings page `/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard&subpath=%2Ffree-listings%2Fedit`
2. Do a change
3. Try navigating away with `< Edit free listings` button 
4. Cancel the prompt
5. Click in browser's chrome to close the tab or reload the page
6. :red_circle: You should get the confirmation prompt

Reply:
Which version are you testing? ITs working fine for me

Reply:
WC 6.5

https://user-images.githubusercontent.com/5908855/177748956-e0570094-e0fb-4f3b-9ac6-cd6d2e56431d.mov


WC 6.7

https://user-images.githubusercontent.com/5908855/177749839-9cfd5657-8ccb-4ee0-ac1c-55d7be008d26.mov